code/datums/cleaning.dm

```diff
@@ -0,0 +1,60 @@
+//For handling standard click-to-clean items like soap and mops.
+
+//text1, text2, and text3 carry strings that, when pieced together by this proc, make the cleaning messages.
+//text1 and text2 pertain to the start message, i.e. "User begins to [clean] the floor[ with the mop.]"
+//text3 pertains to the end message, i.e. "You [finished mopping] the floor."
+
+/atom/proc/cleaning_act(atom/target, mob/user, atom/cleaner, cleanspeed = 50, text1 = "clean", text2 = " with [src].", text3 = "clean")
+	var/cmag_cleantime = 50 //The cleaning time for cmagged objects is locked to this, for balance reasons
+
+	if(user.client && (target in user.client.screen)) //You can't clean items you're wearing for technical reasons
+		to_chat(user, "<span class='notice'>You need to take that [target.name] off before cleaning it.</span>")
+		return
+
+	if(HAS_TRAIT(target, TRAIT_CMAGGED))
+		text1 = "clean the ooze off"
+		text3 = "clean the ooze off"
+		cleanspeed = cmag_cleantime
+
+	if(istype(target, /obj/effect/decal/cleanable) || istype(target, /obj/effect/rune))
+		text1 = "scrub out"
+		text3 = "scrub out"
```

Comment:
With the change from `target` to `src`, these are technically `istype(src, ...)`, which are bad. What you can do is:
```dm
/obj/effect/decal/cleanable/cleaning_act(mob/user, atom/cleaner, cleanspeed = 50, text1 = "scrub out", text2 = " with [src].", text3 = "scrub out"
	..()
```
At first this looks like it does nothing but notice the param replacement of `text1` and `text3` with 
`"scrub out"`,